src/main.rs

```diff
@@ -122,7 +122,14 @@ pub fn run_opt(opt: &Opt) -> Result<bool, Error> {
 
     println!("{}", svdata);
 
-    Ok(all_pass)
+    let ret: Option<SvData>;
+    if all_pass {
+        ret = Some(svdata);
+    } else {
+        ret = None;
+    }
+
+    Ok((all_pass, ret))
```

Comment:
As per comment on line 62, this can be simplified to `Ok(ret)`. The boolean information in `all_pass` is already captured in the `Option` type of `ret`.